mne_connectivity/spectral/epochs.py

```diff
@@ -387,60 +510,30 @@ def _epoch_spectral_connectivity(
         sig_idx = slice(None, None)
 
     # compute tapered spectra
-    x_t = list()
-    this_psd = list()
-    for this_data in data:
-        if mode in ("multitaper", "fourier"):
-            if isinstance(this_data, _BaseSourceEstimate):
-                _mt_spectra_partial = partial(_mt_spectra, dpss=window_fun, sfreq=sfreq)
-                this_x_t = this_data.transform_data(
-                    _mt_spectra_partial,
-                    idx=sig_idx,
-                    tmin_idx=tmin_idx,
-                    tmax_idx=tmax_idx,
-                )
-            else:
-                this_x_t, _ = _mt_spectra(
-                    this_data[sig_idx, tmin_idx:tmax_idx], window_fun, sfreq
-                )
-
-            if mt_adaptive:
-                # compute PSD and adaptive weights
-                _this_psd, weights = _psd_from_mt_adaptive(
-                    this_x_t, eigvals, freq_mask, return_weights=True
-                )
-
-                # only keep freqs of interest
-                this_x_t = this_x_t[:, :, freq_mask]
-            else:
-                # do not use adaptive weights
-                this_x_t = this_x_t[:, :, freq_mask]
-                if mode == "multitaper":
-                    weights = np.sqrt(eigvals)[np.newaxis, :, np.newaxis]
-                else:
-                    # hack to so we can sum over axis=-2
-                    weights = np.array([1.0])[:, None, None]
-
-                if accumulate_psd:
-                    _this_psd = _psd_from_mt(this_x_t, weights)
-        else:  # mode == 'cwt_morlet'
-            if isinstance(this_data, _BaseSourceEstimate):
-                cwt_partial = partial(cwt, Ws=wavelets, use_fft=True, mode="same")
-                this_x_t = this_data.transform_data(
-                    cwt_partial, idx=sig_idx, tmin_idx=tmin_idx, tmax_idx=tmax_idx
-                )
-            else:
-                this_x_t = cwt(
-                    this_data[sig_idx, tmin_idx:tmax_idx],
-                    wavelets,
-                    use_fft=True,
-                    mode="same",
-                )
-            _this_psd = (this_x_t * this_x_t.conj()).real
-
-        x_t.append(this_x_t)
+    if spectrum_computed:  # use existing spectral info
+        # XXX: Will need to distinguish time-resolved spectra here if support added
+        # Select signals & freqs of interest (flexible indexing for optional tapers dim)
+        x_t = np.array(data)[:, sig_idx, ..., freq_mask]
+        if weights is None:  # also assumes no tapers dim
+            x_t = np.expand_dims(x_t, axis=2)  # CSD construction expects a tapers dim
+            weights = np.array([1.0])[:, None, None]
```

Comment:
Equivalent I think but a little more explicit about resulting dimensionality
```suggestion
            weights = np.ones((1, 1, 1))
```

Reply:
Have changed there and also in `_compute_spectra()`